test/cgroups.bats

```diff
@@ -3,8 +3,8 @@
 load helpers
 
 function setup() {
-	newconfig="$TESTDIR/config.json"
 	setup_test
```

Comment:
Hmm. Since `setup_test` sets `$TESTDIR`, what do we have as its value before that?

If that's empty/unset, then maybe `set -u` would help

Reply:
Indeed it helps

```console
[kir@kir-rhat x]$ cat a.bats 
set -u

function setup_test() {
	TESTDIR=$(mktemp -d)
}

function setup() {
	dir="$TESTDIR"/xxx
	setup_test
}

function cleanup() {
	rm -f "$TESTDIR"
}

@test 1 {
	echo 1 $dir >&3
}
[kir@kir-rhat x]$ bats -t  a.bats 
1..1
not ok 1 1
# (from function `setup' in test file a.bats, line 8)
#   `dir="$TESTDIR"/xxx' failed
# /tmp/bats-run-5n7oz0/bats.1269087.src: line 8: TESTDIR: unbound variable
```

but it might uncover other issues (which, I guess, is good).

With `set -u` in place, we can remove the idiomatic check for `test_setup`.

Reply:
So IIUC your suggestion is to add a `set -u` to `helpers.bash` to force it on every bats test script ?

I wonder how a sustainable idea that is ? With such a setup I *think* tests like `[ -n "$1" ]` and `[ -z "$2" ]` and evaluations like `foo "$@"` will start failing, despite looking like idiomatically correct shell expressions. They will need to be replaced with tautological expressions like `[ -n "${1:-}" ]` and `foo "${@:-}"`.

But I guess one way to find out if whether it is sustainable, is to try it out in practice. So I'll make that change and check if the fallout zone is of the size I can clean out.

Reply:
Yeah, the shell expansion on $[1-9] will cause error if the parameter is not passed ($@ will be safe instead).
Having something able to catch more general errors (like the set -u) would be interesting indeed.

Reply:
@fgiudici I'm testing now the `set -u` suggestion from @kolyshkin in practice, trying to go through the errors, understand where unbound arguments are due to intentional optionality, and fix those to work with `set -u` as well. It does not look too bad at first glance.

If I manage to get that working with a reasonably small change set, I will update this PR with that. I think it might be still a good idea to maybe put `set -u` behind an extra variable, so folks can do `make STRICT_MODE=true localintegration`, or the other way around if it's an opt-out... WDYT?

Reply:
@klihub , nice!
The STRICT_MODE option is interesting... but I'm not sure it could be really useful.  In the end I think makes sense to have all the tests to comply to the STRICT_MODE or none of them: is there a reasonable use case to disable (or enable if disabled by default) the STRICT_MODE?

Reply:
@fgiudic, @kolyshkin I've done a first round of set-u-ization of the helpers and tests. Although I tried to be careful enough to tell apart validly unset variables from strictly required ones, I might have made mistakes there. If you guys have the time/interest to look through the topmost patch (I'll update it if I find something more) to see if you spot something obviously wrong, any eyeballing is much appreciated.